File: Scripts/volatile_and_force_compiler_task/ackermann-function/c++/ackermann-function.cpp
#include <iostream>

unsigned int ackermann(unsigned int m, unsigned int n) {
  if (!m) return n + 1;
  if (!n) return ackermann(m - 1, 1);
  return ackermann(m - 1, ackermann(m, n - 1));
}

int executeTask(int i) {

  int aVar = 0;
  if (i % 2 == 0)
	aVar = 1;
  else
	aVar = 2;

  for (unsigned int m = 0; m <= 4 - aVar; ++m) {
    for (unsigned int n = aVar; n <= 3; ++n) {
    	ackermann(m, n);
    }
 }
  return i;
}

int main() {

  	volatile int r = 1;
  	for (int i = 0; i < 10000000; ++i) {
  		r = executeTask(i);
	}

}

File: Scripts/volatile_and_force_compiler_task/ackermann-function/r/ackermann-function.r
ackermann <- function(m, n) {
  if ( m == 0 ) {
    n+1
  } else if ( n == 0 ) {
    ackermann(m-1, 1)
  } else {
    ackermann(m-1, ackermann(m, n-1))
  }
}

executeTask <- function(i) {

  if ( i %% 2 == 0 ) { 
	aVar = 1
  } else {
	aVar = 2
  }

  for ( m in 0:(4 - aVar) ) {
    for ( n in aVar:3 ) {
      ackermann(m, n)
    }
  }

	return(i)
}
aVar = 0
r = 1

for ( i in 0:10000000 ) {
	r = executeTask(i)	
}
